src/flux/transition.coffee

```diff
@@ -0,0 +1,46 @@
+_ = require 'underscore'
+flux = require 'flux-react'
+DestinationHelper = require '../helpers/routes-and-destinations'
+
+TransitionActions = flux.createActions [
+  'load'
+  'reset'
+  '_get'
+]
+
+
+# Transition Store only loads into memory paths that are 'pushed'
+# onto the react-router history.
+# This means that the back button will only track the routes that are
+# 'transitioned' to and not those that 'replace' location,
+# as is the case with router.replaceWith
+#
+# TLDR: Use router.transitionTo or Router.Link for routes you want
+# in History and back button.  Only use router.replaceWith when you
+# want that route to be ignored in History and back.
+
+TransitionStore = flux.createStore
+  actions: _.values(TransitionActions)
+  _local: []
+
+  load: (obj) ->
+    obj.type ?= 'push'
```

Comment:
**Minor:** rather than change the object that was passed in, what about something like:

```coffee
load: ({path, type}) ->
  type ?= 'push'
  @_local.push(path) if type is 'push'
```